src/store.ts

```diff
@@ -15,31 +15,94 @@ import userReducer, { initialUserState } from './auth/ducks/reducers';
 import { ThunkDispatch } from '@reduxjs/toolkit';
 import thunk from 'redux-thunk';
 import tokenService from './auth/token';
+import apiClient, { ApiExtraArgs } from './api/apiClient';
+import {
+  UserLeaderboardWeeklyReducerState,
+  UserLeaderboardMonthlyReducerState,
+  UserLeaderboardYearlyReducerState,
+  UserLeaderboardAllTimeReducerState,
+} from './containers/volunteer-leaderboard/ducks/types';
+import { LeaderboardItemActions } from './components/leaderboard/ducks/actions';
+import userLeaderboardWeeklyReducer, {
+  initialUserLeaderboardWeeklyState,
+} from './containers/volunteer-leaderboard/ducks/weeklyReducer';
+import userLeaderboardMonthlyReducer, {
+  initialUserLeaderboardMonthlyState,
+} from './containers/volunteer-leaderboard/ducks/monthlyReducer';
+import userLeaderboardYearlyReducer, {
+  initialUserLeaderboardYearlyState,
+} from './containers/volunteer-leaderboard/ducks/yearlyReducer';
+import userLeaderboardAllTimeReducer, {
+  initialUserLeaderboardAllTimeState,
+} from './containers/volunteer-leaderboard/ducks/allTimeReducer';
+import {
+  TeamLeaderboardWeeklyReducerState,
+  TeamLeaderboardMonthlyReducerState,
+  TeamLeaderboardYearlyReducerState,
+  TeamLeaderboardAllTimeReducerState,
+} from './containers/team-leaderboard/ducks/types';
+import teamLeaderboardWeeklyReducer, {
+  initialTeamLeaderboardWeeklyState,
+} from './containers/team-leaderboard/ducks/weeklyReducer';
+import teamLeaderboardMonthlyReducer, {
+  initialTeamLeaderboardMonthlyState,
+} from './containers/team-leaderboard/ducks/monthlyReducer';
+import teamLeaderboardYearlyReducer, {
+  initialTeamLeaderboardYearlyState,
+} from './containers/team-leaderboard/ducks/yearlyReducer';
+import teamLeaderboardAllTimeReducer, {
+  initialTeamLeaderboardAllTimeState,
+} from './containers/team-leaderboard/ducks/allTimeReducer';
 
 export interface C4CState {
   authenticationState: UserAuthenticationReducerState;
+  userLeaderboardWeeklyState: UserLeaderboardWeeklyReducerState;
+  userLeaderboardMonthlyState: UserLeaderboardMonthlyReducerState;
+  userLeaderboardYearlyState: UserLeaderboardYearlyReducerState;
+  userLeaderboardAllTimeState: UserLeaderboardAllTimeReducerState;
+  teamLeaderboardWeeklyState: TeamLeaderboardWeeklyReducerState;
+  teamLeaderboardMonthlyState: TeamLeaderboardMonthlyReducerState;
+  teamLeaderboardYearlyState: TeamLeaderboardYearlyReducerState;
+  teamLeaderboardAllTimeState: TeamLeaderboardAllTimeReducerState;
```

Comment:
Seems like some nesting would be helpful, having a leaderboardState that then contains user and team states